src/indexes.rs

```diff
@@ -628,8 +636,9 @@ impl Index {
     /// # index.delete().await.unwrap().wait_for_completion(&client, None, None).await.unwrap();
     /// # });
     /// ```
-    pub async fn get_primary_key(&self) -> Result<Option<String>, Error> {
-        Ok(self.fetch_info().await?.primaryKey)
+    pub async fn get_primary_key(&mut self) -> Result<Option<String>, Error> {
+        self.fetch_info().await?;
+        Ok(self.primary_key.to_owned())
```

Comment:
Now that we actually store the primary key somewhere.
I think it would be better to return a reference to the primary key instead of an owned version and let the user clone it if they need to. What do you think?

```suggestion
    pub async fn get_primary_key(&mut self) -> Result<Option<&str>, Error> {
        self.fetch_info().await?;
        Ok(self.primary_key.as_ref())
```

Reply:
Oh yeah, that makes sense.